mbuild/formats/hoomd3_forcefield.py

```diff
@@ -0,0 +1,402 @@
+        )
+        hoomd_forcefield.append(harmonic_bond)
+    if structure.angle_types:
+        print("Processing harmonic angles")
+        harmonic_angle = _init_hoomd_angles(structure, ref_energy=ref_energy)
+        hoomd_forcefield.append(harmonic_angle)
+    if structure.dihedral_types:
+        print("Processing periodic torsions")
+        periodic_torsions = _init_hoomd_dihedrals(
+            structure, ref_energy=ref_energy
+        )
+        hoomd_forcefield.append(periodic_torsions)
+    if structure.rb_torsion_types:
+        print("Processing RB torsions")
+        rb_torsions = _init_hoomd_rb_torsions(structure, ref_energy=ref_energy)
+        hoomd_forcefield.append(rb_torsions)
+
+    return snapshot, hoomd_forcefield, ref_values
+
+
+def _init_hoomd_lj(structure, nl, r_cut=1.2, ref_distance=1.0, ref_energy=1.0):
+    """ LJ parameters """
+    # Identify the unique atom types before setting
+    atom_type_params = {}
+    for atom in structure.atoms:
+        if atom.type not in atom_type_params:
+            atom_type_params[atom.type] = atom.atom_type
+
+    # Set the hoomd parameters for self-interactions
+    lj = hoomd.md.pair.LJ(nlist=nl)
+    for name, atom_type in atom_type_params.items():
+        lj.params[(name, name)] = dict(
+            sigma=atom_type.sigma / ref_distance,
+            epsilon=atom_type.epsilon / ref_energy,
+        )
+        lj.r_cut[(name, name)] = r_cut
+
+    # Cross interactions, mixing rules, NBfixes
+    all_atomtypes = sorted(atom_type_params.keys())
+    for a1, a2 in itertools.combinations_with_replacement(all_atomtypes, 2):
+        nb_fix_info = atom_type_params[a1].nbfix.get(a2, None)
+        # nb_fix_info = (rmin, eps, rmin14, eps14)
+        if nb_fix_info is None:
+            # No nbfix means use mixing rule to find cross-interaction
+            if structure.combining_rule == "lorentz":
+                sigma = (
+                    atom_type_params[a1].sigma + atom_type_params[a2].sigma
+                ) / (2 * ref_distance)
+                epsilon = (
+                    (
+                        atom_type_params[a1].epsilon
+                        * atom_type_params[a2].epsilon
+                    )
+                    / ref_energy ** 2
+                ) ** 0.5
+            elif structure.combining_rule == "geometric":
+                sigma = (
+                    (atom_type_params[a1].sigma * atom_type_params[a2].sigma)
+                    / ref_distance ** 2
+                ) ** 0.5
+                epsilon = (
+                    (
+                        atom_type_params[a1].epsilon
+                        * atom_type_params[a2].epsilon
+                    )
+                    / ref_energy ** 2
+                ) ** 0.5
+            else:
+                raise ValueError(
+                    "Mixing rule {} ".format(structure.combining_rule)
+                    + "not supported, use lorentz"
```

Comment:
Lorentz or geometric, right?

Reply:
Yes. I think this is code that @mikemhenry wrote and I ported to v3. I will fix the string.

I would be great in future developments if there were some core reusable code to compute the mixing rules that hoomd and other backends could rely on.